lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/VortexDriver.java

```diff
@@ -0,0 +1,180 @@
+    this.vortexUserCode = vortexUserCode;
+    this.vortexThreadPool = vortexThreadPool;
+    this.evalMem = evalMem;
+    this.evalNum = evalNum;
+    this.evalCores = evalCores;
+  }
+
+  /**
+   * Driver started.
+   */
+  public final class StartHandler implements EventHandler<StartTime> {
+    @Override
+    public void onNext(final StartTime startTime) {
+      // Initial Evaluator Request
+      evaluatorRequestor.submit(EvaluatorRequest.newBuilder()
+          .setNumber(evalNum)
+          .setMemory(evalMem)
+          .setNumberOfCores(evalCores)
+          .build());
+
+      // Run the user code as a separate thread
+      // todo: handle user code exceptions
+      vortexUserThread.execute(new Runnable() {
+        @Override
+        public void run() {
+          vortexUserCode.start(vortexThreadPool);
+          vortexManager.terminate();
+        }
+      });
+    }
+  }
+
+  /**
+   * Container allocated.
+   */
+  public final class AllocatedEvaluatorHandler implements EventHandler<AllocatedEvaluator> {
+    @Override
+    public void onNext(final AllocatedEvaluator allocatedEvaluator) {
+      LOG.log(Level.INFO, "Container allocated");
+
+      final Configuration executorConfiguration = ExecutorConfiguration.CONF
+          .set(ExecutorConfiguration.NUM_OF_THREADS, evalCores) // NUM_OF_THREADS = evalCores
+          .build();
+
+      final Configuration contextConfiguration = ContextConfiguration.CONF
+              .set(ContextConfiguration.IDENTIFIER, allocatedEvaluator.getId() + "_context")
+              .set(ContextConfiguration.ON_MESSAGE, LocalExecutor.class)
+              .set(ContextConfiguration.ON_SEND_MESSAGE, LocalExecutor.class)
+              .build();
+
+      allocatedEvaluator.submitContext(Configurations.merge(executorConfiguration, contextConfiguration));
+    }
+  }
+
+  /**
+   * Evaluator up and running.
+   */
+  public final class ActiveContextHandler implements EventHandler<ActiveContext> {
+    @Override
+    public void onNext(final ActiveContext activeContext) {
+      LOG.log(Level.INFO, "Evaluator up and running");
+      vortexManager.taskletExecutorAllocated(new DefaultTaskletExecutor(commandSender, activeContext));
+    }
+  }
+
+  /**
+   * Message received.
+   */
+  public final class ContextMessageHandler implements EventHandler<ContextMessage> {
+    @Override
+    public void onNext(final ContextMessage contextMessage) {
+      final String executorId = contextMessage.getId();
+      final ReportTaskletResult result = (ReportTaskletResult)SerializationUtils.deserialize(contextMessage.get());
+      LOG.log(Level.INFO, "Completed Tasklet id: " + result.getTaskletId());
+      LOG.log(Level.INFO, "Completed Tasklet result: " + result.getResult());
+      vortexManager.taskletCompleted(executorId, result.getTaskletId(), result.getResult());
+    }
+  }
+
+  /**
+   * Evaluator preempted.
+   */
+  public final class FailedEvaluatorHandler implements EventHandler<FailedEvaluator> {
```

Comment:
New JIRA for this:
Distinguish different types of FailedEvaluator in Vortex
https://issues.apache.org/jira/browse/REEF-501